### DISCUSSION TITLE
Saving to BMP v3 with alpha channel, without weird single color fill.

### DISCUSSION THREAD

**Discussion by futuremotiondev** [ImageMagick/ImageMagick#5555] [Help]
I am trying to convert a PNG to BMP (Version 3) with alpha channel via PowerShell script. I am using the following command [found here](http://www.imagemagick.org/discourse-server/viewtopic.php?t=28733&sid=4936322be135a229e261024b42b97696&start=15) with the latest version of ImageMagick:

`magick $CurFile -define bmp3:alpha=true bmp3:$DestFile`

This works, but the output file has a solid color fill "behind" the alpha channel. The best way to describe is through imagery:

Original PNG:

![dllhost_JgQBdRPp9H](https://user-images.githubusercontent.com/33441569/189477155-df4f282a-2b46-4483-aa24-5889cffc85aa.png)

BMP v3 with alpha channel (Saved via Photoshop CC 2022):

![dllhost_Br9QOKXU4M](https://user-images.githubusercontent.com/33441569/189477164-c0984c65-2071-4cea-8154-30238ae0ec63.png)

BMP v3 with alpha channel (Saved via ImageMagick):

![dllhost_KZfGZCOe9A](https://user-images.githubusercontent.com/33441569/189477170-358ee2f7-c1c2-4655-8b3a-1a3fac068309.png)

From what I can tell the BMP file saved with ImageMagick has a solid color fill the same width and height as the alpha channel's bounding box. It doesn't matter much because when the alpha is recognized and utilized this fill is invisible. It's only apparent when being viewed with an application that doesn't respect the alpha channel. However, I'd really like to avert this processing as it makes quick previews in both Windows explorer and Photoshop look bad. I want to emulate Photoshop's output as much as possible as they seem to get this right. Here's a better example:

Screenshot of Photoshop when opening the file generated via IM:

![Icon Photoshop 2](https://user-images.githubusercontent.com/33441569/189477093-7bfd77da-c9df-4b33-a6f3-cb676ca37773.png)

Screenshot of Photoshop when opening the file saved via Photoshop:

![Icon Photoshop 1](https://user-images.githubusercontent.com/33441569/189477106-6ee26944-cf32-4175-bae9-73bd30d0ada9.png)

Hopefully this all makes sense now. I don't want IM to touch the RGB channels at all, just the alpha.

I've attached a zip of:

1. The original PNG
2. The BMPv3 file saved from Photoshop
3. The BMPv3 file saved from an ImageMagick conversion

If anyone knows how to skip or reverse this processing, it would be greatly appreciated!

Thanks,
Jay

**Comment by futuremotiondev** [comment]
Forgot to attach the zip. Here it is.

[PNGTesting.zip](https://github.com/ImageMagick/ImageMagick/files/9539928/PNGTesting.zip)

**Comment by snibgo** [comment]
>  I don't want IM to touch the RGB channels at all, just the alpha.

But the problem is that in "Driving32x32 Original File.png" some pixels are transparent white, and some are transparent blue.

I think you want to change this, so all the transparent pixels are transparent white.
```
magick "Driving32x32 Original File.png" -background White -alpha Background -define bmp3:alpha=true bmp3:x.bmp
```

**Comment by snibgo** [comment]
In case it needs saying ...
> From what I can tell the BMP file saved with ImageMagick has a solid color fill the same width and height as the alpha channel's bounding box.

No, that isn't true. The problem is in "Driving32x32 Original File.png", as you will see if you view it in any program that can ignore the alpha channel.

**Comment by fmw42** [comment]
Works fine for me on IM 7.1.0.48 Q16 Mac OSX Monterey:

![original](https://user-images.githubusercontent.com/13458218/190711726-8237bc50-4724-4945-a5fe-9273d4b0b39f.png)


```
magick original.png -define bmp3:alpha=true bmp3:x3.png
magick convert x3.png -alpha off x3b.png
```

x3:

![x3](https://user-images.githubusercontent.com/13458218/190711818-baf70b91-ee90-473c-870e-dd5023091124.png)

X3b:

![x3b](https://user-images.githubusercontent.com/13458218/190711845-d80be2b9-8c9e-44e5-bf32-8101adfe9516.png)

**Comment by dlemstra** [comment]
I just pushed a patch to improve the rendering of the bmp in the next version (7.1.0-58).

**Comment by snibgo** [comment]
See also https://github.com/ImageMagick/ImageMagick/discussions/7876 , where the patch seems to cause other problems.
